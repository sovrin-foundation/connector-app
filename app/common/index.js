// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
