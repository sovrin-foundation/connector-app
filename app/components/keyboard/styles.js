// @flow

import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  isBiggerThanShortDevice,
  deviceHeight,
  MEDIUM_DEVICE,
} from '../../common/styles/constant'
import { PUSH_NOTIFICATION_PERMISSION } from '../../push-notification/type-push-notification'

const ROW_MARGIN = isBiggerThanShortDevice ? 36 : 15
const FIRST_ROW_MARGIN = isBiggerThanShortDevice ? 24 : 8

const ROW_HEIGHT =
  Platform.OS === 'ios' ? 36 : deviceHeight < MEDIUM_DEVICE ? 27 : 36

const FONT_SIZE = isBiggerThanShortDevice ? 28 : 26
export const INPUT_RANGE = [0, 0.5, 1]
export const OUTPUT_RANGE =
  Platform.OS === 'ios'
    ? isBiggerThanShortDevice ? [28, 29, 30] : [26, 30, 34]
    : deviceHeight < MEDIUM_DEVICE ? [28, 30, 32] : [26, 30, 34]

export const styles = StyleSheet.create({
  container: {
    marginHorizontal: 32, // TODO: update style guide to reflect new constant
    marginBottom: 5,
  },
  firstRow: {
    height: ROW_HEIGHT,
    marginTop: FIRST_ROW_MARGIN,
  },
  row: {
    height: ROW_HEIGHT,
    marginTop: ROW_MARGIN,
  },
  number: {
    fontSize: FONT_SIZE,
    fontWeight: '500', // TODO: update style guide to reflect new constant
  },
})
