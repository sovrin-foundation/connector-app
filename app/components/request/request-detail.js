// @flow
import React, { PureComponent } from 'react'
import { Container, CustomText, CustomView } from '../../components'
import RequestDetailText from './request-detail-text'
import RequestDetailAvatars from './request-detail-avatars'
import type { RequestDetailProps } from './type-request'

export default class RequestDetail extends PureComponent<
  RequestDetailProps,
  void
> {
  render() {
    const { testID } = this.props
    return (
      <Container useNativeDriver hCenter testID={`${testID}-text-container`}>
        <Container bottom testID={`${testID}-text-message-container`}>
          <RequestDetailText
            title={this.props.title}
            message={this.props.message}
            testID={testID}
          />
        </Container>
        <Container testID={`${testID}-text-container-avatars`}>
          <RequestDetailAvatars senderLogoUrl={this.props.senderLogoUrl} />
        </Container>
      </Container>
    )
  }
}
