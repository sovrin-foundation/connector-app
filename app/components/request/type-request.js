// @flow
import type { ReactNavigation, CustomError } from '../../common/type-common'

export type noop = () => any

export type RequestDetailTextPropsText = {
  title: string,
  message: string,
  testID: string,
}

export type RequestDetailTextProps = RequestDetailTextPropsText

export type RequestDetailAvatarProps = {
  senderLogoUrl?: ?string,
}

export type RequestDetailProps = RequestDetailTextProps &
  RequestDetailAvatarProps

export const ResponseType = {
  accepted: 'accepted',
  rejected: 'rejected',
  none: 'none',
}

// TODO: Need to move this to one place for each store such as invitation, qr connection, & authentication request
export type ResponseTypes = $Keys<typeof ResponseType>

export type RequestProps = RequestDetailProps & {
  onAction: (response: ResponseTypes) => void,
  showErrorAlerts: boolean,
  isTouchIdEnabled: boolean,
  pushNotificationPermissionAction: boolean => void,
  invitationError?: ?CustomError,
  senderName: string,
} & ReactNavigation

export type RequestState = {
  disableAccept: boolean,
}
