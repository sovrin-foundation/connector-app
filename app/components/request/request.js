// @flow
import React, { PureComponent } from 'react'
import { View, Alert } from 'react-native'
import firebase from 'react-native-firebase'
import { bindActionCreators } from 'redux'
import { Container, TouchId } from '../../components'
import { connect } from 'react-redux'
import { TOUCH_ID_MESSAGE, TOUCH_ID_NOT_AVAILABLE, noop } from '../../common'
import RequestDetail from './request-detail'
import FooterActions from '../footer-actions/footer-actions'
import type { RequestProps, RequestState, ResponseTypes } from './type-request'
import { captureError } from '../../services/error/error-handler'
import { lockAuthorizationRoute } from '../../common/route-constants'
import type { Store } from '../../store/type-store'
import { pushNotificationPermissionAction } from '../../push-notification/push-notification-store'
import {
  ERROR_ALREADY_EXIST,
  ERROR_INVITATION_RESPONSE_FAILED,
  ERROR_ALREADY_EXIST_TITLE,
} from '../../api/api-constants'

export class Request extends PureComponent<RequestProps, RequestState> {
  state = {
    disableAccept: false,
  }

  onAccept = () => {
    this.setState({ disableAccept: true })
    // Move these values to enum, we are not doing it now because of TODO in type file
    return this.onAction('accepted')
  }

  onDecline = () => {
    // Move these values to enum, we are not doing it now because of TODO in type file
    return this.onAction('rejected')
  }

  onSuccessfulAuthorization = (response: ResponseTypes) => {
    this.props.onAction(response)
  }

  onDuplicateConnectionError = () => {
    this.onSuccessfulAuthorization('rejected')
  }

  onAvoidAuthorization = () => {
    /**
     * User can choose to avoid entering passcode either by canceling
     * or by pressing back button and choose to ignore passcode
     * then once user comes back, we need to re-enable button
     * so that user can press accept or decline button again
     */
    this.setState({ disableAccept: false })
  }

  onTouchIdFailed = (response: ResponseTypes) =>
    this.props.navigation.navigate(lockAuthorizationRoute, {
      onSuccess: () => this.onSuccessfulAuthorization(response),
      onAvoid: () => this.onAvoidAuthorization(),
    })

  authenticate = (response: ResponseTypes) => {
    return TouchId.isSupported()
      .then(result => TouchId.authenticate(TOUCH_ID_MESSAGE))
      .then(() => this.onSuccessfulAuthorization(response))
      .catch(() => this.onTouchIdFailed(response))
  }

  checkIfTouchIdEnabled(response: ResponseTypes) {
    if (this.props.isTouchIdEnabled) {
      return this.authenticate(response)
    } else {
      this.onTouchIdFailed(response)
    }
  }

  onAction = (response: ResponseTypes) => {
    return firebase
      .messaging()
      .requestPermission()
      .then(() => {
        this.props.pushNotificationPermissionAction(true)
        this.checkIfTouchIdEnabled(response)
      })
      .catch(error => {
        // astute readers will notice that we are calling authenticate
        // in success and fail both, so we can move it outside of promise
        // but we need to authenticate after user take allow/deny action
        // for push notification, after user allows/denies push notification
        // we need to authenticate user either with TouchId or passcode
        // unfortunately React-Native's (ES6's as well) Promise implementation
        // does not have finally block
        this.checkIfTouchIdEnabled(response)
        // TODO: we did not get push token
        // now what should we do?
        captureError(error, this.props.showErrorAlerts)
      })
  }

  componentDidUpdate(prevProps: RequestProps) {
    // When Accept invitation errors out we are re-enabling accept button. Giving user the option to retry.
    if (
      this.props.invitationError !== prevProps.invitationError &&
      this.props.invitationError
    ) {
      const isDuplicateConnection =
        this.props.invitationError.code === ERROR_ALREADY_EXIST.code
      const errorMessage = isDuplicateConnection
        ? `${this.props.invitationError.message}${this.props.senderName}`
        : ERROR_INVITATION_RESPONSE_FAILED
      const okAction = isDuplicateConnection
        ? this.onDuplicateConnectionError
        : noop
      const errorTitle = isDuplicateConnection
        ? ERROR_ALREADY_EXIST_TITLE
        : null

      Alert.alert(
        errorTitle,
        errorMessage,
        [{ text: 'Ok', onPress: okAction }],
        {
          cancelable: false,
        }
      )

      this.setState({
        disableAccept: false,
      })
    }
  }

  render() {
    const { title, message, senderLogoUrl, testID }: RequestProps = this.props

    return (
      <Container>
        <Container fifth>
          <RequestDetail
            title={title}
            message={message}
            senderLogoUrl={senderLogoUrl}
            testID={testID}
          />
        </Container>
        <FooterActions
          disableAccept={this.state.disableAccept}
          disableDeny={this.state.disableAccept}
          onAccept={this.onAccept}
          onDecline={this.onDecline}
          logoUrl={senderLogoUrl}
          testID={testID}
          useColorPicker={true}
        />
      </Container>
    )
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      pushNotificationPermissionAction,
    },
    dispatch
  )

const mapStateToProps = ({ lock }: Store) => {
  return {
    isTouchIdEnabled: lock.isTouchIdEnabled,
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Request)
