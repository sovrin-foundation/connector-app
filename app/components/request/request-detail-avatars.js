// @flow
import React, { PureComponent } from 'react'
import { StyleSheet, Image } from 'react-native'
import { View as AnimatedView } from 'react-native-animatable'
import { Container, CustomView, UserAvatar, Avatar } from '../../components'
import { getConnectionLogo } from '../../store/connections-store'
import type { RequestDetailAvatarProps } from './type-request'
import { OFFSET_4X, OFFSET_1X } from '../../common/styles'
import type { ImageSource } from '../../common/type-common'

export default class RequestDetailAvatars extends PureComponent<
  RequestDetailAvatarProps,
  void
> {
  renderAvatarWithSource = (avatarSource: number | ImageSource) => (
    <Avatar
      medium
      shadow
      src={avatarSource}
      testID={'invitation-text-avatars-invitee'}
    />
  )

  render() {
    const senderAvatar = getConnectionLogo(this.props.senderLogoUrl)

    return (
      <AnimatedView
        animation="bounceInUp"
        delay={300}
        duration={1500}
        testID={'invitation-text-container-avatars-animation'}
        accessible={true}
        accessibilityLabel={'invitation-text-container-avatars-animation'}
      >
        <CustomView center spaceBetween>
          <CustomView
            row
            vCenter
            spaceBetween
            style={[styles.avatarsContainer]}
            testID={'invitation-text-avatars-container'}
          >
            <UserAvatar>{this.renderAvatarWithSource}</UserAvatar>
            <Image
              style={styles.forwardArrow}
              source={require('../../images/connectDots.png')}
            />
            <Avatar
              medium
              shadow
              src={senderAvatar}
              testID={'invitation-text-avatars-inviter'}
            />
          </CustomView>
        </CustomView>
      </AnimatedView>
    )
  }
}

const styles = StyleSheet.create({
  avatarsContainer: {
    marginVertical: OFFSET_4X,
  },
  forwardArrow: {
    width: 60,
    height: 8,
    marginHorizontal: OFFSET_1X / 2,
  },
})
