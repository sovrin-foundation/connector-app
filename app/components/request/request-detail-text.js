// @flow
import React, { PureComponent } from 'react'
import { StyleSheet } from 'react-native'
import { View as AnimatedView } from 'react-native-animatable'
import { CustomText, CustomView } from '../../components'
import { OFFSET_1X, OFFSET_3X } from '../../common/styles'
import { noop } from '../../common'
import type { RequestDetailTextProps } from './type-request'

export default class RequestDetailText extends PureComponent<
  RequestDetailTextProps,
  void
> {
  render() {
    const { testID } = this.props
    return (
      <AnimatedView
        testID={`${testID}-text-container-animation`}
        accessible={true}
        accessibilityLabel={`${testID}-text-container-animation`}
        animation="bounceInDown"
        duration={1500}
        delay={100}
      >
        <CustomView center testID={`${testID}-text-container-message-title`}>
          <CustomText
            testID={`${testID}-text-title`}
            h4
            center
            thick
            bg="fifth"
            style={[styles.textTitle]}
            testID={`${testID}-text-container-title`}
          >
            {this.props.title}
          </CustomText>
          <CustomText
            h5
            center
            bold
            bg="fifth"
            style={[styles.textMessage]}
            testID={`${testID}-text-container-message`}
          >
            {this.props.message}
          </CustomText>
        </CustomView>
      </AnimatedView>
    )
  }
}

const styles = StyleSheet.create({
  textMessage: {
    margin: OFFSET_1X,
  },
  textTitle: {
    marginVertical: OFFSET_1X,
    marginHorizontal: OFFSET_3X,
    lineHeight: OFFSET_3X,
  },
})
