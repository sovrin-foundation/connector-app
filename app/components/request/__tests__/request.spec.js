// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import TouchId from 'react-native-touch-id'
import firebase from 'react-native-firebase'

import { Request } from '../request'
import type { RequestProps, ResponseTypes } from '../type-request'
import { color } from '../../../common/styles/constant'
import { lockAuthorizationRoute } from '../../../common/route-constants'
import { getNavigation, getStore } from '../../../../__mocks__/static-data'

describe('<Request />', () => {
  let store
  let request
  let requestComponent
  let tree
  const accepted = 'accepted'
  const rejected = 'rejected'
  const defaultProps = {
    title: 'Hi Test User',
    message: 'Enterprise A agent wants to connect with you',
    senderLogoUrl: 'https://image.url',
    senderName: 'Enterprise A agent',
    onAction: jest.fn(),
    showErrorAlerts: false,
    testID: 'request',
    isTouchIdEnabled: true,
    pushNotificationPermissionAction: jest.fn(),
  }
  let navigation

  beforeAll(() => {
    store = getStore()
  })

  beforeEach(() => {
    navigation = getNavigation({ onSuccess: jest.fn() })

    // onAction = jest.fn()
    request = renderer.create(
      <Provider store={store}>
        <Request
          {...defaultProps}
          onAction={defaultProps.onAction}
          navigation={navigation}
          showErrorAlerts={defaultProps.showErrorAlerts}
        />
      </Provider>
    )
    tree = request.toJSON()
    requestComponent = request.root.findByType(Request).instance
  })

  it('should match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('TouchId and calls onAction if Connect/Deny button is pressed', async () => {
    const touchIdAuth = await requestComponent.onAccept()
    await touchIdAuth
    expect(defaultProps.onAction).toBeCalled()
    expect(defaultProps.onAction).toHaveBeenCalledWith(accepted)
  })

  it('redirect user to lock authorization page if TouchId fails', async () => {
    TouchId.authenticate.mockImplementation(message => Promise.reject())
    await requestComponent.authenticate(accepted)
    expect(navigation.navigate).toHaveBeenCalledWith(lockAuthorizationRoute, {
      onSuccess: expect.any(Function),
      onAvoid: expect.any(Function),
    })
  })

  it('call authenticate even if user allow push notification', async () => {
    await requestComponent.onAction(accepted)
    expect(defaultProps.onAction).toHaveBeenCalledWith(accepted)
  })

  it('call authenticate even if user does not allow push notification', async () => {
    firebase
      .messaging()
      .requestPermission.mockImplementation(() => Promise.reject())
    await requestComponent.onAction(accepted)
    expect(defaultProps.onAction).toHaveBeenCalledWith(accepted)
  })
})
