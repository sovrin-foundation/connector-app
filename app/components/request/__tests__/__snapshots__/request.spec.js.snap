// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Request /> should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    fifth={true}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
  >
    <View
      accessibilityLabel="request-text-container"
      accessible={true}
      hCenter={true}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
      testID="request-text-container"
      useNativeDriver={true}
    >
      <View
        accessibilityLabel="request-text-message-container"
        accessible={true}
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
        testID="request-text-message-container"
      >
        <Animatable.View
          accessibilityLabel="request-text-container-animation"
          accessible={true}
          animation="bounceInDown"
          delay={100}
          duration={1500}
          testID="request-text-container-animation"
        >
          <View
            accessibilityLabel="request-text-container-message-title"
            accessible={true}
            center={true}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              }
            }
            testID="request-text-container-message-title"
          >
            <Text
              accessibilityLabel="request-text-container-title"
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 20,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "#777777",
                  },
                  Object {
                    "fontWeight": "800",
                  },
                  Object {
                    "textAlign": "center",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  Object {
                    "lineHeight": 30,
                    "marginHorizontal": 30,
                    "marginVertical": 10,
                  },
                ]
              }
              testID="request-text-container-title"
            >
              Hi Test User
            </Text>
            <Text
              accessibilityLabel="request-text-container-message"
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontFamily": "Lato",
                  },
                  Object {
                    "backgroundColor": "#ffffff",
                    "color": "#777777",
                  },
                  Object {
                    "fontWeight": "bold",
                  },
                  Object {
                    "textAlign": "center",
                  },
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  null,
                  Object {
                    "margin": 10,
                  },
                ]
              }
              testID="request-text-container-message"
            >
              Enterprise A agent wants to connect with you
            </Text>
          </View>
        </Animatable.View>
      </View>
      <View
        accessibilityLabel="request-text-container-avatars"
        accessible={true}
        style={
          Object {
            "flex": 1,
          }
        }
        testID="request-text-container-avatars"
      >
        <Animatable.View
          accessibilityLabel="invitation-text-container-avatars-animation"
          accessible={true}
          animation="bounceInUp"
          delay={300}
          duration={1500}
          testID="invitation-text-container-avatars-animation"
        >
          <View
            center={true}
            spaceBetween={true}
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessibilityLabel="invitation-text-avatars-container"
              accessible={true}
              row={true}
              spaceBetween={true}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginVertical": 40,
                }
              }
              testID="invitation-text-avatars-container"
              vCenter={true}
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                pointerEvents="box-only"
                style={undefined}
                testID={undefined}
              >
                <View
                  accessibilityLabel="invitation-text-avatars-invitee"
                  accessible={true}
                  imageShadow="imageShadow"
                  style={
                    Object {
                      "borderRadius": 38,
                      "elevation": 3,
                      "shadowColor": "rgba(0, 0, 0, 0.25)",
                      "shadowOffset": Object {
                        "height": 3,
                        "width": 0,
                      },
                      "shadowOpacity": 1,
                      "shadowRadius": 8,
                    }
                  }
                  testID="invitation-text-avatars-invitee"
                >
                  <Image
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    hitSlop={undefined}
                    nativeID={undefined}
                    onError={undefined}
                    onLayout={undefined}
                    onLoad={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    source={
                      Object {
                        "uri": "/var/application/DocumentDir/user-avatar.jpeg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": 76,
                          "width": 76,
                        },
                        Object {
                          "borderRadius": 38,
                        },
                      ]
                    }
                    testID={undefined}
                  />
                </View>
              </View>
              <Image
                source={1}
                style={
                  Object {
                    "height": 8,
                    "marginHorizontal": 5,
                    "width": 60,
                  }
                }
              />
              <View
                accessibilityLabel="invitation-text-avatars-inviter"
                accessible={true}
                imageShadow="imageShadow"
                style={
                  Object {
                    "borderRadius": 38,
                    "elevation": 3,
                    "shadowColor": "rgba(0, 0, 0, 0.25)",
                    "shadowOffset": Object {
                      "height": 3,
                      "width": 0,
                    },
                    "shadowOpacity": 1,
                    "shadowRadius": 8,
                  }
                }
                testID="invitation-text-avatars-inviter"
              >
                <Image
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  nativeID={undefined}
                  onError={undefined}
                  onLayout={undefined}
                  onLoad={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  source={
                    Object {
                      "uri": "https://image.url",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 76,
                        "width": 76,
                      },
                      Object {
                        "borderRadius": 38,
                      },
                    ]
                  }
                  testID={undefined}
                />
              </View>
            </View>
          </View>
        </Animatable.View>
      </View>
    </View>
  </View>
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "rgba(119, 119, 119, 1.0)",
      }
    }
  >
    <View
      fifth={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          customColor={
            Object {
              "backgroundColor": "rgba(119, 119, 119, 1.0)",
            }
          }
          row={true}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 0,
                      "marginRight": 0,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel="request-deny"
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                    ]
                  }
                  testID="request-deny"
                  tvParallaxProperties={undefined}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "rgba(119, 119, 119, 1.0)",
                        },
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "paddingVertical": 17,
                          },
                        ],
                        false,
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#ffffff",
                            },
                            Object {
                              "fontSize": 14,
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            false,
                          ],
                        ]
                      }
                    >
                      Deny
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 0,
                      "marginRight": 0,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel="request-accept"
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Array [
                        Object {
                          "borderLeftColor": "#ffffff",
                          "borderLeftWidth": 0.4,
                        },
                      ],
                    ]
                  }
                  testID="request-accept"
                  tvParallaxProperties={undefined}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "rgba(119, 119, 119, 1.0)",
                        },
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "paddingVertical": 17,
                          },
                          Object {
                            "borderLeftColor": "#ffffff",
                            "borderLeftWidth": 0.4,
                          },
                        ],
                        false,
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#ffffff",
                            },
                            Object {
                              "fontSize": 14,
                            },
                            undefined,
                            undefined,
                            Object {
                              "fontWeight": "bold",
                            },
                            undefined,
                            false,
                          ],
                        ]
                      }
                    >
                      Connect
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
